### PR TITLE
refactored using DB connection

### PR DESCRIPTION
**Pull request by vadym-uskov**
Added new library to the common_api.robot:
Library    DatabaseLibrary

To install this library just use a command: pip install -U robotframework-databaselibrary

### CODE REVIEW

File: resources/common/common_api.robot [hunk 1]
```diff
@@ -15,6 +16,12 @@ ${api_timeout}                 60
 ${default_password}            change123
 ${default_allow_redirects}     true
 ${default_auth}                ${NONE}
+# *** DB VARIABLES ***
+${db_host}         127.0.0.1
+${db_name}         eu-docker
+${db_password}     secret
+${db_port}         3306
+${db_user}         spryker
```
Comment (OKrivtsova) [inline resources/common/common_api.robot:24]: please add 'default' for variable names - different envs may have different variables

File: resources/common/common_api.robot [hunk 2]
```diff
@@ -1122,6 +1129,33 @@ Save value to a variable:
...
+    ...
+    ...    ``[Setup]    Connect To Database    pymysql    ${db_name}    ${db_user}    ${db_password}    ${db_host}    ${db_port}``
+    ...
+    ...    ``Save value from DB to a variable:    select registration_key from spy_customer where customer_reference = '${user_reference_id}'    confirmation_key``
+    ...
+    ...    ``[Teardown]    Disconnect From Database``
+    ...
+    ...    ``Save value from DB to a variable:`` should be called after you connect to the DB. You can set connection in [Setup] of your current test case and also don't forget to disconnect from the DB in the [Teardown] section.
+    [Arguments]    ${sql_query}    ${variable_name}
+    ${var_value} =    Query    ${sql_query}
```
Comment (OKrivtsova) [inline resources/common/common_api.robot:1150]: is there a way to handle errors? if the query returned more than 1 variable? can we show custom error or is ootb error shown is enough?
if the query returns more than 1 row, does variable get assigned an array?

File: resources/common/common_api.robot [hunk 3]
```diff
@@ -1122,6 +1129,33 @@ Save value to a variable:
...
+    [Documentation]    This keyword saves any value which you receive from DB using SQL query ``${sql_query}`` to a test variable called ``${variable_name}``. 
+    ...
+    ...    It can be used to save a value returned by any query into a custom test variable.
+    ...    This variable, once created, can be used during the specific test where this keyword is used and can be re-used by the keywords that follow this keyword in the test. 
+    ...    It will not be visible to other tests.
+    ...    NOTE: Make sure that you expect only 1 value from DB, you can also check your query via external SQL tool.
+    ...
+    ...    *Examples:*
+    ...
+    ...    ``[Setup]    Connect To Database    pymysql    ${db_name}    ${db_user}    ${db_password}    ${db_host}    ${db_port}``
```
Comment (OKrivtsova) [inline resources/common/common_api.robot:1142]: this does snot need to be in example.
this needs to be in the notes if we need them at all.

what is the risk of adding this and disconnect to the body of this keyword? we do ot normally need the DB connection for a prolonged time so we can do it like this:
```

[Arguments]    ${sql_query}    ${variable_name}
    Connect To Database    pymysql    ${db_name}    ${db_user}    ${db_password}    ${db_host}    ${db_port}
    ${var_value} =    Query    ${sql_query}
    Disconnect From Database
    ${var_value}=    Convert To String    ${var_value}
    ${var_value}=    Replace String    ${var_value}    '   ${EMPTY}
    ${var_value}=    Replace String    ${var_value}    ,   ${EMPTY}
    ${var_value}=    Replace String    ${var_value}    (   ${EMPTY}
    ${var_value}=    Replace String    ${var_value}    )   ${EMPTY}
    Set Test Variable    ${${variable_name}}    ${var_value}  
    [Return]    ${variable_name}

```

File: tests/api/b2c/glue/customer_endpoints/customer_confirmation/negative.robot [hunk 4]
```diff
@@ -1,54 +1,56 @@
...
-#     ...    AND    I send a DELETE request:    /customers/${userId}
-#     ...    AND    Response status code should be:    204
-#     ...    AND    Response reason should be:    No Content
+Customer_confirmation_with_already_used_confirmation_key
+    [Setup]    Run Keywords    I send a POST request:    /customers    {"data":{"type":"customers","attributes":{"firstName":"${yves_third_user_first_name}","lastName":"${yves_third_user_last_name}","gender":"${yves_third_user_gender_male}","salutation":"${yves_third_user_salutation}","email":"${yves_third_user_first_name}.${yves_third_user_last_name}${random}@spryker.com","password":"${yves_third_user_password}","confirmPassword":"${yves_third_user_password}","acceptedTerms":True}}}
+    ...    AND    Response status code should be:    201
+    ...    AND    Save value to a variable:    [data][id]    user_reference_id
+    ...    AND    Connect To Database    pymysql    ${db_name}    ${db_user}    ${db_password}    ${db_host}    ${db_port}
+    ...    AND    Save value from DB to a variable:    select registration_key from spy_customer where customer_reference = '${user_reference_id}'    confirmation_key
```
Comment (OKrivtsova) [inline tests/api/b2c/glue/customer_endpoints/customer_confirmation/negative.robot:43]: so if we add connect and disconnect into the keyword - we do not need connect above and disconnect below

File: tests/api/b2c/glue/customer_endpoints/customer_confirmation/positive.robot [hunk 5]
```diff
@@ -1,22 +1,25 @@
...
-#     ...    AND    Response status code should be:    204
-#     ...    AND    Response reason should be:    No Content
\ No newline at end of file
+
+Customer_confirmation
+    [Setup]    Run Keywords    I send a POST request:    /customers    {"data":{"type":"customers","attributes":{"firstName":"${yves_third_user_first_name}","lastName":"${yves_third_user_last_name}","gender":"${yves_third_user_gender_male}","salutation":"${yves_third_user_salutation}","email":"${yves_third_user_first_name}.${yves_third_user_last_name}${random}@spryker.com","password":"${yves_third_user_password}","confirmPassword":"${yves_third_user_password}","acceptedTerms":True}}}
+    ...    AND    Response status code should be:    201
+    ...    AND    Save value to a variable:    [data][id]    user_reference_id
+    ...    AND    Connect To Database    pymysql    ${db_name}    ${db_user}    ${db_password}    ${db_host}    ${db_port}
+    ...    AND    Save value from DB to a variable:    select registration_key from spy_customer where customer_reference = '${user_reference_id}'    confirmation_key
```
Comment (OKrivtsova) [inline tests/api/b2c/glue/customer_endpoints/customer_confirmation/positive.robot:16]: same here

File: resources/common/common_api.robot [hunk 6]
```diff
@@ -1122,6 +1129,33 @@ Save value to a variable:
     Set Test Variable    ${${name}}    ${var_value}  
     [Return]    ${name}
 
+Save value from DB to a variable:
```
Comment (OKrivtsova) [inline resources/common/common_api.robot:1132]: rename to 
Save the result of a SELECT DB query to a variable:

File: resources/common/common_api.robot (hunk 1 above)
Comment (vadym-uskov) [reply resources/common/common_api.robot:24]: fixed

File: resources/common/common_api.robot (hunk 6 above)
Comment (vadym-uskov) [reply resources/common/common_api.robot:1132]: fixed

File: resources/common/common_api.robot (hunk 3 above)
Comment (vadym-uskov) [reply resources/common/common_api.robot:1142]: fixed, added connections to the keyword

File: tests/api/b2c/glue/customer_endpoints/customer_confirmation/negative.robot (hunk 4 above)
Comment (vadym-uskov) [reply tests/api/b2c/glue/customer_endpoints/customer_confirmation/negative.robot:43]: fixed

File: tests/api/b2c/glue/customer_endpoints/customer_confirmation/positive.robot (hunk 5 above)
Comment (vadym-uskov) [reply tests/api/b2c/glue/customer_endpoints/customer_confirmation/positive.robot:16]: fixed

File: README.md [hunk 7]
```diff
@@ -10,6 +10,10 @@
 2. Install RequestsLibrary: `pip install robotframework-requests`
 3. Install JSONLibrary: `pip install -U robotframework-jsonlibrary`
 
+### For DB testing installation requires Robot Framework, [DatabaseLibrary](https://github.com/franz-see/Robotframework-Database-Library)
+1. Install [Robot Framework](https://github.com/robotframework/robotframework/blob/master/INSTALL.rst)
+2. Install DatabaseLibrary: `pip install -U robotframework-databaselibrary`
```
[the remark this answers is not in the document]
Comment (vadym-uskov) [reply README.md]: fixed

**Review by OKrivtsova — APPROVED** [review]

**Pull request merged**